unit_test.py

```diff
@@ -411,7 +459,121 @@ def test_services_table(self):
+                    {
+                        "peerASN": 65002,
+                        "peerAddress": "192.168.11.101/32"
+                    },
+                    {
+                        "peerASN": 65002,
+                        "peerAddress": "192.168.11.102/32"
+                    }
+                    ],
+                    "serviceSelector": {
+                    "matchExpressions": [
+                        {
+                        "key": "CiliumAdvertiseAllServices",
+                        "operator": "DoesNotExist"
+                        }
+                    ]
+                    }
+                }
+                ]
+            }
+            },
+            {
+            "apiVersion": "cilium.io/v2alpha1",
+            "kind": "CiliumBGPPeeringPolicy",
+            "metadata": {
+                "creationTimestamp": "2023-01-20T04:49:55Z",
+                "generation": 1,
+                "name": "rack-2",
+                "resourceVersion": "977",
+                "uid": "f264c06b-4e65-44d1-9050-e87c3e52a39f"
+            },
+            "spec": {
+                "nodeSelector": {
+                "matchLabels": {
+                    "rack_id": "2"
+                }
+                },
+                "virtualRouters": [
+                {
+                    "exportPodCIDR": true,
+                    "localASN": 56001,
+                    "neighbors": [
+                    {
+                        "peerASN": 65002,
+                        "peerAddress": "192.168.11.103/32"
+                    },
+                    {
+                        "peerASN": 65002,
+                        "peerAddress": "192.168.11.104/32"
+                    }
+                    ],
+                    "serviceSelector": {
+                    "matchExpressions": [
+                        {
+                        "key": "CiliumAdvertiseAllServices",
+                        "operator": "DoesNotExist"
+                        }
+                    ]
+                    }
+                }
+                ]
+            }
+            }
+        ],
+        "kind": "List",
+        "metadata": {
+            "resourceVersion": ""
+        }
+        }
+        """)
+        """Raise a Calico Exception and hide the Kube-Router POD , that triggers testing for Cilium"""
+        with (
+            patch('kubernetes.client.CustomObjectsApi.get_cluster_custom_object', MagicMock(side_effect=Exception("Test"))),
+            patch('kubernetes.client.CustomObjectsApi.list_cluster_custom_object', MagicMock(return_value=CiliumBGPPeeringPolicies)),
+            patch('kubernetes.client.CoreV1Api.list_pod_for_all_namespaces', MagicMock(return_value=client.V1PodList(api_version="1", items=pods[0:1]))),
+        ):
+            build = VkaciBuilTopology(
+                VkaciEnvVariables(self.vars), ApicMethodsMock())
+            build.update()
+            asn = build.get_cluster_as()

```

Comment:
The cluster AS detection is being executed twice in the unit tests which was not required.